libs/designsystem/src/lib/components/checkbox/checkbox.component.scss

```diff
@@ -1,52 +1,102 @@
-  --background-checked: #{get-color('primary')};
-  --border-color-checked: #{get-color('primary')};
-}
+    &:active {
+      --background-checked: #{get-color('success-shade')};
+      --border-color-checked: #{get-color('success-shade')};
+    }
+  }
 
-.secondary {
-  --checkmark-color: #{get-color('secondary-contrast')};
-  --background-checked: #{get-color('secondary')};
-  --border-color-checked: #{get-color('secondary')};
-}
+  &.attention-level2 ion-checkbox {
+    --checkmark-color: #{get-color('white')};
+    --background-checked: #{get-color('black')};
+    --border-color-checked: #{get-color('black')};
 
-.tertiary {
-  --checkmark-color: #{get-color('tertiary-contrast')};
-  --background-checked: #{get-color('tertiary')};
-  --border-color-checked: #{get-color('tertiary')};
-}
+    &:active {
+      --checkmark-color: #{get-color('white-shade')};
+      --background-checked: #{get-color('black-tint')};
+      --border-color-checked: #{get-color('black-tint')};
+    }
+  }
 
-.warning {
-  --checkmark-color: #{get-color('warning-contrast')};
-  --background-checked: #{get-color('warning')};
-  --border-color-checked: #{get-color('warning')};
-}
+  &.error ion-checkbox {
+    --border-color: #{get-color('danger')};
+  }
 
-.success {
-  --checkmark-color: #{get-color('success-contrast')};
-  --background-checked: #{get-color('success')};
-  --border-color-checked: #{get-color('success')};
-}
+  &.disabled {
+    color: #{get-text-color('semi-dark')};
 
-.danger {
-  --checkmark-color: #{get-color('danger-contrast')};
-  --background-checked: #{get-color('danger')};
-  --border-color-checked: #{get-color('danger')};
+    ion-checkbox {
+      opacity: 1;
+      --checkmark-color: #{get-color('semi-dark')};
+      --background: #{get-color('semi-light')};
+      --background-checked: #{get-color('semi-light')};
+      --border-color: #{get-color('medium')};
+      --border-color-checked: #{get-color('semi-light')};
+    }
+  }
+
+  ion-checkbox {
+    --checkmark-width: #{size('xxxs')};
+    --background: #{get-color('white')};
+    --border-width: 1px;
+    --border-color: #{get-color('semi-dark')};
+    --border-radius: #{$border-radius};
+
+    // Making clickable area bigger without making icon bigger
+    --size: #{$fat-finger-size};
+    padding: #{($fat-finger-size - $checkbox-icon-size) / 2};
+
+    &::part(container) {
+      padding: size('xxxs');
+    }
+
+    &:focus-within::part(container) {
+      @media (hover: hover) {
+        box-shadow: 0 0 0 1px var(--kirby-background-color), 0 0 0 2px #{$focus-ring-color};
+        border-radius: $border-radius;
+      }
+    }
+    &:active {
+      --background: #{get-color('white-shade')};
+    }
+  }
+
+  &.has-label {
+    div {
+      position: relative;
+      display: inline-flex;
+      align-items: center;
+      vertical-align: top;
+      padding-right: size('s');
+    }
+    ion-checkbox {
+      position: static;
+      margin-right: size('xxxs');
+    }
+  }
```

Comment:
Shouldn't all this be handled by conditionally wrapping checkbox in a kirby-label (hence ion-label)?

Reply:
I think this solution is simpler. Adding a wrapping kirby-label will only add more complication without any gains.
Semantically a kirby-label and ion-label do not add anything, as they do not contain an html label, and the Ionic getAriaLabel can't read a wrapping ion-label without a parent ion-item.